openlibrary/i18n/__init__.py

```diff
@@ -101,6 +103,34 @@ def load_translations(lang):
     if os.path.exists(mo_path):
         return Translations(open(mo_path, 'rb'))
 
+
+def get_ol_locale():
+    """
+    Gets the locale from the cookie and -if not found- sets it to the
+    highest priority language of the Accept-Language header, that's also part
+    of the Open Library supported languages.
+    """
+    locale_cookie = web.cookies().get('i18n_code')
+    if locale_cookie:
+        ol_locale = locale_cookie
+    else:
+        browser_accepted_langs = web.ctx.env.get('HTTP_ACCEPT_LANGUAGE')
+        if browser_accepted_langs:
+            # Parse the accepted language header string into a list
+            languages = browser_accepted_langs.split(',')
+            # Remove the quality-value weights and keep only the locales
+            accepted_langs = [lang.partition(';')[0] for lang in languages]
+            # The default locale is 'en' unless we support a browser-suggested language
+            ol_locale = 'en'
+            for lang in accepted_langs:
+                if lang in OL_SUPPORTED_LANGAGES:
+                    ol_locale = lang
+                    break
+        else:
+            ol_locale = 'en'
+        web.setcookie('i18n_code', ol_locale, secure=False)
```

Comment:
I don't think we want to set this cookie unless the user explicitly clicks the button. If the user changes their browser settings, and they haven't explicitly set the language, we DO want OL to use the right language. (This also makes testing easier :P )

```suggestion
```

Reply:
Done